src/activities/morse_code/MorseCode.qml

```diff
@@ -0,0 +1,567 @@
+                 "A" : ".-", "B" : "-...", "C" : "-.-.", "D" : "-..",  "E" : ".", "F" : "..-.", "G" : "--.",
+                 "H" : "....", "I" : "..", "J" : ".---", "K" : "-.-",  "L" : ".-..","M" : "--","N" : "-.",
+                 "O" : "---", "P" : ".--.", "Q" : "--.-", "R" : ".-.",  "S" : "...","T" : "-", "U" : "..-",
+                 "V" : "...-", "W" : ".--",  "X" : "-..-",  "Y" : "-.--","Z" : "--..","1" : ".----","2" : "..---",
+                 "3" : "...--",  "4" : "....-", "5" : ".....", "6" : "-....",  "7" : "--...",  "8" : "---..",
+                 "9" : "----." , "0" : "-----"
+                };
+
+                var code=""
+                for( var i = 0; i < str.length; i++ )
+                {
+                    var temp = table[ str.charAt(i) ];
+                    if( temp )
+                    {
+                        if( "*" === temp )
+                        {
+                            temp = " ";
+                        }
+                        code += temp + " ";
+                    }
+                    else code += " ";
+                }
+                morseConverter.last=str
+                return code
+            }
+            onAlphaChanged: morse = alpha2morse(alpha)
+            onMorseChanged: alpha = morse2alpha(morse)
+
+        }
+
+        Column {
+            id: column
+            anchors.fill: parent
+            anchors.topMargin: 10
+            Image {
+                id: tux
+                source:url+"tux.svg"
+                fillMode: Image.PreserveAspectCrop
+                width: 100 * ApplicationInfo.ratio
+                height: 100 * ApplicationInfo.ratio
+
+                anchors.bottomMargin: 200
+                anchors.leftMargin: 75
+                anchors.horizontalCenter: parent.horizontalCenter
+            }
+
+            GCText {
+                id: questionLabel
+                anchors.horizontalCenter: parent.horizontalCenter
+                text: items.questionValue ? items.questionText.arg(items.questionValue) : ''
+                color: 'white'
+                fontSizeMode: Text.Fit
+                Rectangle {
+                    z: -1
+                    border.color: 'black'
+                    border.width: 1
+                    anchors.centerIn: parent
+                    width: parent.width * 1.1
+                    height: parent.height
+                    opacity: 0.8
+                    gradient: Gradient {
+                        GradientStop { position: 0.0; color: "#000" }
+                        GradientStop { position: 0.9; color: "#666" }
+                        GradientStop { position: 1.0; color: "#AAA" }
+                    }
+                    radius: 10
+                }
+            }
+            Item { // Just a margin
+                width: 1
+                height: 5 * ApplicationInfo.ratio
+            }
+            TextInput {
+                id: textInput
+                x: parent.width / 2.1
```

Comment:
why don't you use "anchors.horizontalCenter: parent.horizontalCenter" instead of manually putting a x?